netshare/utils/field.py

```diff
@@ -44,7 +59,14 @@ def normalize(self, x):
                 f"Dimension is {x.shape[-1]}. Expected dimension is {self.dim_x}"
             )
         if self.log1p_norm:
+            # If set log1p_norm, we need to re-calculate the min_x and max_x after doing
+            # log1p. Since the denormalizing step would first do min-max denomalization
+            # and then expm1. Using stale min_x and max_x would cause the synthetic
+            # value larger than expected and is possible to have np.inf
             x = np.log1p(x)
+            self.min_x = np.log1p(self.min_x)
+            self.max_x = np.log1p(self.max_x)
+
```

Comment:
When activating log1p_norm, we need also to norm the min and max. Because in denoamlizing stage, the generated data will first be denomalized by min-max denoamlizagion, then expm1. Using a unnormalized min max may made the result larger then expected and sometimes may cause it become inf.

Reply:
That's a good catch!
Can you please move this code to the __init__?
`normalize` should be a pure function: I.e. we don't expect it to change any internal value while operating. I.e. someone can call it multiple times and that's fine. Here, you're changing members inside `self`